tests/src/testApplicationFactory.cc

```diff
@@ -0,0 +1,114 @@
+#define BOOST_TEST_MODULE TestApplicationFactory
+// Only after defining the name include the unit test header.
+#include "ApplicationFactory.h"
+#include "ReferenceTestApplication.h"
+
+#include <boost/test/included/unit_test.hpp>
+
+using namespace boost::unit_test_framework;
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_CASE(testNoFactoryNoInstance) {
+  try {
+    (void)ApplicationBase::getInstance();
+    BOOST_ERROR("ApplicationBase::getInstance() did not throw as expected.");
+  }
+  catch(ChimeraTK::logic_error& e) {
+    std::cout << "Exception message debug printout: " << e.what() << std::endl;
+  }
+
+  try {
+    (void)ApplicationFactoryBase::getApplicationInstance();
+    BOOST_ERROR("ApplicationFactoryBase::getInstance() did not throw as expected.");
+  }
+  catch(ChimeraTK::logic_error& e) {
+    std::cout << "Exception message debug printout: " << e.what() << std::endl;
+  }
+}
+
+BOOST_AUTO_TEST_CASE(testFactory) {
```

Comment:
Please add "chrome bar" separators between test cases for better readability.

Reply:
done